salt/_renderers/metalk8s_kubernetes.py

```diff
@@ -210,11 +211,21 @@ def _handle_v1_namespace(obj, kubeconfig, context):
 
 @handle('v1', 'Secret')
 def _handle_v1_secret(obj, kubeconfig, context):
+    # FIXME #1183: this is a compatibility band-aid for the vendored salt module
+    if 'data' in obj:
+        data = {
+            key: base64.b64decode(value)
+            for key, value in obj['data'].items()
+        }
+    elif 'stringData' in obj:
```

Comment:
Not sure about these next lines: should we just die in a fire if there is no `data` key?

Reply:
No it looks good, at least it should work for both cases (don't know if all upstream manifests encode their secrets).